base/special/exp.jl

```diff
@@ -1,65 +1,200 @@
-# Based on FDLIBM http://www.netlib.org/fdlibm/e_exp.c
-# which is made available under the following licence
-
-## Copyright (C) 2004 by Sun Microsystems, Inc. All rights reserved. Permission
-## to use, copy, modify, and distribute this software is freely granted,
-## provided that this notice is preserved.
-
-# Method
-# 1. Argument reduction: Reduce x to an r so that |r| <= 0.5*ln(2). Given x,
-#    find r and integer k such that
-#       x = k*ln(2) + r,  |r| <= 0.5*ln(2).
-#    Here r is represented as r = hi - lo for better accuracy.
-#
-# 2. Approximate exp(r) by a special rational function on [0, 0.5*ln(2)]:
-#       R(r^2) = r*(exp(r)+1)/(exp(r)-1) = 2 + r*r/6 - r^4/360 + ...
-#
-#    A special Remez algorithm on [0, 0.5*ln(2)] is used to generate a
-#    polynomial to approximate R.
-#
-#    The computation of exp(r) thus becomes
-#                       2*r
-#       exp(r) = 1 + ----------
-#                     R(r) - r
-#                          r*c(r)
-#              = 1 + r + ----------- (for better accuracy)
-#                         2 - c(r)
-#    where
-#       c(r) = r - (P1*r^2  + P2*r^4  + ... + P5*r^10 + ...).
-#
-# 3. Scale back: exp(x) = 2^k * exp(r)
+ℯ = 2.71828182845904523536
```

Comment:
I do not see why this isn't declared `const`. Is it worth adding a comment to this line?

Reply:
This was a janky hack to work around the fact that `MathConstants.ℯ` doesn't exist when this loads. If there's a better way to do this, I would look forward to it.

Reply:
Since you're only using it as a "key" in `Val`-based dispatch on the specific kernels/constants, you could just swap out that one case for a symbol — i.e. use `Val{:e}` or `Val{:ℯ}` instead. It'd be a little bit asymmetric with the `Val{2}` and `Val{10}` that you're using for the other two cases, but it might actually be nicer than seeing `::Val{2.71828182845904523536}` written out in e.g. `methods(Base.Math.expb_kernel)`.

Reply:
Makes sense